tests/Feature/Api/V1Test.php

```diff
@@ -178,23 +178,88 @@ public function testGetAchievementOfTheWeek(): void
             ]);
     }
 
-    // public function testGetAchievementsEarnedBetween(): void
-    // {
-    //     // TODO
-    //
-    //     $this->get($this->apiUrl('GetAchievementsEarnedBetween'))
-    //         ->assertSuccessful()
-    //         ->assertExactJson([]);
-    // }
-    //
-    // public function testGetAchievementsEarnedOnDay(): void
-    // {
-    //     // TODO
-    //
-    //     $this->get($this->apiUrl('GetAchievementsEarnedOnDay'))
-    //         ->assertSuccessful()
-    //         ->assertExactJson([]);
-    // }
+    public function testGetAchievementsEarnedBetweenEmptyResponse(): void
+    {
+        $this->get($this->apiUrl('GetAchievementsEarnedBetween'))
+            ->assertSuccessful()
+            ->assertExactJson([]);
+    }
+
+    public function testGetAchievementsEarnedBetween(): void
+    {
+        /** @var System $system */
+        $system = System::factory()->create();
+        /** @var Game $game */
+        $game = Game::factory()->create(['ConsoleID' => $system->ID]);
+        /** @var Achievement $achievement */
+        $achievement = Achievement::factory()->published()->create(['GameID' => $game->ID, 'Points' => 100]);
+        /** @var PlayerAchievement $unlock */
+        $unlock = PlayerAchievement::factory()->create(['AchievementID' => $achievement->ID, 'User' => $this->user->User]);
+
+        $this->get(
+            $this->apiUrl('GetAchievementsEarnedBetween', [
+                'u' => $this->user->User,
+                'f' => Carbon::now()->subDay()->startOfDay()->unix(),
+                't' => Carbon::now()->addDay()->endOfDay()->unix(),
+            ])
+        )
+            ->assertSuccessful()
+            ->assertJson([
+                [
+                    'AchievementID' => $achievement->ID,
+                    'ConsoleName' => $system->Name,
+                    'CumulScore' => 100,
+                    'Date' => $unlock->Date->format('Y-m-d H:i:s'),
+                    'Description' => $achievement->Description,
+                    'GameID' => $game->ID,
+                    'GameIcon' => '/Images/000001.png',
```

Comment:
should you be expecting `$game->Badge` in case the factory generates something else in the future?